bookmarks/forms.py

```diff
@@ -30,7 +30,15 @@ class Meta:
         widgets = {
             'name': forms.TextInput(attrs={'class': 'category_name_widget'}),
         }
+    
+    def clean(self):
+        cleaned_data = super().clean()
+        name = self.cleaned_data.get("name", )
 
+        if BookmarkCategory.objects.filter(user=self.instance.user, name=name).exists():
+            raise forms.ValidationError("This name already exists for a bookmark category")
```

Comment:
This error message could be improved. Maybe "You have already created a bookmark category with that name"